### PR TITLE
Store "your code may have changed" dismissal in localStorage

### PR DESCRIPTION
**Pull request by maddiedierker**
We've gotten feedback from users that the "Your code may have changed" banner/alert was too intrusive (more details in the Jira ticket). This is what the banner looks like:
Screen Shot 2020-10-02 at 10 25 42 AM

Before, this banner was displayed every time you edited code while your app was running. Now, if the user dismisses this banner, we will store it in localStorage. [Video of new behavior](https://drive.google.com/file/d/1qb2nomg5nqvEFYpYmxKWP82I4nJnaNjK/view?usp=sharing)

## Links

- [STAR-1225](https://codedotorg.atlassian.net/browse/STAR-1225)

## Testing story

Added unit tests for this feature.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/test/unit/StudioAppTest.js [hunk 1]
```diff
@@ -16,42 +16,41 @@ import project from '@cdo/apps/code-studio/initApp/project';
...
       document.body.removeChild(codeWorkspaceDiv);
       document.body.removeChild(containerDiv);
     });
 
-    beforeEach(() => {
-      stubRedux();
-      registerReducers(commonReducers);
-    });
-    afterEach(restoreRedux);
-
```
Comment (maddiedierker) [inline apps/test/unit/StudioAppTest.js:54]: all changes above this line are just clean up. there were 3 separate beforeEach and afterEach statements for the same scope of tests, so i condensed them.

**Comment by breville** [comment]
Out of curiosity: how do we decide whether to use a cookie vs. local storage vs. a per-user setting in the cloud?

Also, some cookies are env-specific.  Is it worth having similar for local storage?  (Or does it already do that?)

**Comment by maddiedierker** [comment]
i've almost exclusively used local storage, but i still check for similar functionality in the area of the code i'm in. for this PR, i followed the convention from our instructions overlay: https://github.com/code-dot-org/code-dot-org/pull/12191/files

local storage is inherently env-specific, so that comes for free!

**Review by breville — APPROVED** [review]
Not familiar with this code before, but looks like good improvements!

**Pull request merged**